src/Control/Algebra/Named.hs

```diff
@@ -13,6 +13,13 @@ runNamed = runNamedC
 newtype NamedC name (m :: * -> *) a = NamedC { runNamedC :: m a }
   deriving (Applicative, Functor, Monad)
 
-instance (Algebra (eff :+: sig) m, HFunctor eff, HFunctor sig) => Algebra (Named name eff :+: sig) (NamedC name m) where
+type family Lhs a where
+  Lhs (a :+: _) = a
+
+type family Rhs a where
+  Rhs (_ :+: b) = b
+
+instance (Algebra m, Signature m ~ (eff :+: sig), HFunctor eff, HFunctor sig) => Algebra (NamedC name m) where
+  type Signature (NamedC name m) = Named name (Lhs (Signature m)) :+: Rhs (Signature m)
```

Comment:
1. `eff :+: sig` is no longer part of the instance head
2. Despite the type equality constraint in the context, we can’t mention `eff` or `sig` on the rhs of `Signature` because they don’t occur on the lhs; we can therefore only mention `name` and `m`.
3. So we define a couple type families to compute the lhs and rhs of `:+:`. Partiality at the type level is fine, so it doesn’t require the type equality constraint when verifying this type.